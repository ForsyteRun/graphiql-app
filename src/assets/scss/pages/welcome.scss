.welcome-container {
  max-width: 1440px; //TODO
  @include flexColumn(space-between);
  gap: 10rem;
  padding: 0 20px;
}

.welcome-title {
  align-self: flex-start;
}
