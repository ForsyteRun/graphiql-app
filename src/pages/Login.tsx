import { NavLink } from 'react-router-dom';
import { REGISTRATION_ROUTE } from '../constants/route';
import { LoginForm } from '../components';

const Login = () => {
  return (
    <>
      <div className="main page container">
        <div className="authorization">
          <div className="authorization__container">
            <h1>Войти</h1>
            <LoginForm />
          </div>
          <div className="authorization__information">
            <h4>Не зарегистрированы?</h4>
            <p>Создание учетной записи займет не больше минуты.</p>
            <NavLink to={REGISTRATION_ROUTE}>Создать учетную запись</NavLink>
          </div>
        </div>
      </div>
    </>
  );
};

export default Login;
