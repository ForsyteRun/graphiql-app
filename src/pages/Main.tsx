// Main.tsx

import GraphiQL from '../components/GraphiQL';
import { Footer } from '../layout/Footer';
import { Header } from '../layout/Header';

const Main = () => {
  return (
    <div className="tygh">
      <Header />
      <GraphiQL />
      <Footer />
    </div>
  );
};

export { Main };
