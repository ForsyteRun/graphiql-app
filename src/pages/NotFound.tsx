import { NavLink } from 'react-router-dom';
import { WELCOME_ROUTE } from '../constants/route';

const NotFound = () => {
  return (
    <div className="main container">
      <div className="exception">
        <div className="exception__code">
          404
          <span className="exception__code-txt">Ошибка</span>
        </div>
        <div className="exception__title-info">
          <h1 className="exception__title">
            Извините! Мы не смогли найти то, что вы искали.
          </h1>
          <p className="exception__info">Запрашиваемая страница не найдена.</p>
          <div className="exception__links">
            <div className="exception__links-item">
              <NavLink
                to={WELCOME_ROUTE}
                title=""
                className="exception__links-a"
              >
                Перейти на главную страницу
              </NavLink>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default NotFound;
