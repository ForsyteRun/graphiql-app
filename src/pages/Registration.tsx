import { NavLink } from 'react-router-dom';
import { RegistrationForm } from '../components';
import { AUTH_ROUTE } from '../constants/route';

const Registration = () => {
  return (
    <div className="tygh">
      <main className="main container page">
        <div className="authorization">
          <div className="authorization__container">
            <h1>Создать учетную запись</h1>
            <RegistrationForm />
          </div>
          <div className="authorization__information">
            <h4>Преимущества зарегистрированного пользователя:</h4>
            <div className="authorization__login">
              Уже есть учетная запись? Тогда просто жмите
              <NavLink to={AUTH_ROUTE}> Вход</NavLink>.
            </div>
          </div>
        </div>
      </main>
    </div>
  );
};

export default Registration;
