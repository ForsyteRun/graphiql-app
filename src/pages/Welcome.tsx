import { Content, OurTeam } from '../components';

const Welcome = () => {
  return (
    <div className="welcome-container">
      <div className="welcome-title">
        <h1>GraphiQL is the ability to harness the full power of GraphQL</h1>
        <span>make a request to your own universe</span>
      </div>
      <Content />
      <OurTeam />
    </div>
  );
};

export default Welcome;
