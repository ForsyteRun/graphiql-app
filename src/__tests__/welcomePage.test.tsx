import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Welcome } from '../pages';

const intersectionObserverMock = () => ({
  observe: () => null,
  disconnect: jest.fn(),
  unobserve: jest.fn(),
});

window.IntersectionObserver = jest
  .fn()
  .mockImplementation(intersectionObserverMock);

describe('Welcome Page', () => {
  it('should be equal Welcome Page snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should has tree images of developers', () => {
    render(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    );

    const imageElements = screen.getAllByRole('img');

    expect(imageElements.length).toBe(13);
  });

  it('should has tree names of developers', () => {
    render(
      <BrowserRouter>
        <Welcome />
      </BrowserRouter>
    );

    const imageElements = screen.getAllByRole('heading', { level: 3 });

    expect(imageElements.length).toBe(12);
  });
});
