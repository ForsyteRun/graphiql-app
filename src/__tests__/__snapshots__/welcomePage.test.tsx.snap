// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page should be equal Welcome Page snapshot 1`] = `
<DocumentFragment>
  <div
    class="welcome-container"
  >
    <div
      class="welcome-title"
    >
      <h1>
        GraphiQL is the ability to harness the full power of GraphQL
      </h1>
      <span>
        make a request to your own universe
      </span>
    </div>
    <ul
      class="welcome-list"
    >
      <li>
        Efficiency
        <span>
          :
        </span>
        <span>
          Clients get exactly the data they need, no more, no less
        </span>
      </li>
      <li>
        Flexibility
        <span>
          :
        </span>
        <span>
          Clients can request multiple resources in a single query
        </span>
      </li>
      <li>
        Versioning
        <span>
          :
        </span>
        <span>
          No need for versioning as clients have control
        </span>
      </li>
      <li>
        Strongly Typed
        <span>
          :
        </span>
        <span>
          Type checking at the query level
        </span>
      </li>
    </ul>
    <div>
      <h2
        class="ourTeam-title"
      >
        OurTeam
      </h2>
      <div
        class="ourTeam-container"
      >
        <div
          class="team"
        >
          <div
            class="team__info"
          >
            <img
              alt="Katsiaryna"
              class="team__photo"
            />
            <div
              class="team__details"
            >
              <h3
                class="team__name"
              >
                Katsiaryna
              </h3>
              <h4
                class="team__role"
              >
                Developer
              </h4>
              <a
                class="team__github"
                href="https://github.com/Katsiaryna"
                rel="noreferrer"
                target="_blank"
                title="Katsiaryna GitHub"
              >
                Katsiaryna
              </a>
              <div>
                <h4
                  class="team__contribution_title"
                >
                  Contribution
                </h4>
                <ul
                  class="team__contribution_list"
                >
                  <li>
                    JavaScript
                  </li>
                  <li>
                    Python
                  </li>
                  <li>
                    Java
                  </li>
                  <li>
                    C++
                  </li>
                  <li>
                    C#
                  </li>
                  <li>
                    TypeScript
                  </li>
                  <li>
                    Ruby
                  </li>
                  <li>
                    Swift
                  </li>
                  <li>
                    Go
                  </li>
                  <li>
                    Kotlin
                  </li>
                  <li>
                    Rust
                  </li>
                  <li>
                    PHP
                  </li>
                  <li>
                    HTML
                  </li>
                  <li>
                    CSS
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <h4
            class="team__about_title"
          >
            About
          </h4>
          <p
            class="team__about"
          >
            I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
          </p>
        </div>
        <div
          class="team"
        >
          <div
            class="team__info"
          >
            <img
              alt="Forsyte Alex"
              class="team__photo"
            />
            <div
              class="team__details"
            >
              <h3
                class="team__name"
              >
                Forsyte Alex
              </h3>
              <h4
                class="team__role"
              >
                Team Lead
              </h4>
              <a
                class="team__github"
                href="https://github.com/forsyterun"
                rel="noreferrer"
                target="_blank"
                title="Forsyte Alex GitHub"
              >
                forsyterun
              </a>
              <div>
                <h4
                  class="team__contribution_title"
                >
                  Contribution
                </h4>
                <ul
                  class="team__contribution_list"
                >
                  <li>
                    JavaScript
                  </li>
                  <li>
                    Python
                  </li>
                  <li>
                    Java
                  </li>
                  <li>
                    C++
                  </li>
                  <li>
                    C#
                  </li>
                  <li>
                    TypeScript
                  </li>
                  <li>
                    Ruby
                  </li>
                  <li>
                    Swift
                  </li>
                  <li>
                    Go
                  </li>
                  <li>
                    Kotlin
                  </li>
                  <li>
                    Rust
                  </li>
                  <li>
                    PHP
                  </li>
                  <li>
                    HTML
                  </li>
                  <li>
                    CSS
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <h4
            class="team__about_title"
          >
            About
          </h4>
          <p
            class="team__about"
          >
            I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
          </p>
        </div>
        <div
          class="team"
        >
          <div
            class="team__info"
          >
            <img
              alt="nyurasheva"
              class="team__photo"
            />
            <div
              class="team__details"
            >
              <h3
                class="team__name"
              >
                nyurasheva
              </h3>
              <h4
                class="team__role"
              >
                Developer
              </h4>
              <a
                class="team__github"
                href="https://github.com/nyurasheva"
                rel="noreferrer"
                target="_blank"
                title="nyurasheva GitHub"
              >
                nyurasheva
              </a>
              <div>
                <h4
                  class="team__contribution_title"
                >
                  Contribution
                </h4>
                <ul
                  class="team__contribution_list"
                >
                  <li>
                    JavaScript
                  </li>
                  <li>
                    Python
                  </li>
                  <li>
                    Java
                  </li>
                  <li>
                    C++
                  </li>
                  <li>
                    C#
                  </li>
                  <li>
                    TypeScript
                  </li>
                  <li>
                    Ruby
                  </li>
                  <li>
                    Swift
                  </li>
                  <li>
                    Go
                  </li>
                  <li>
                    Kotlin
                  </li>
                  <li>
                    Rust
                  </li>
                  <li>
                    PHP
                  </li>
                  <li>
                    HTML
                  </li>
                  <li>
                    CSS
                  </li>
                </ul>
              </div>
            </div>
          </div>
          <h4
            class="team__about_title"
          >
            About
          </h4>
          <p
            class="team__about"
          >
            I\`m is a Ukrainian singer, producer, and DJ known for his contributions to the pop and electronic music scenes. Born on October 17, 1988, in Chelyabinsk, Russia,
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
